angrmanagement/ui/workspace.py

```diff
@@ -264,7 +264,24 @@ def decompile_current_function(self, view=None):
 
         view.decompile_current_function()
 
-    def decompile_function(self, func, view=None):
+    def position_cursor_on_decomp(self, view, curr_ins):
+        if view is not None and curr_ins is not None:
+            # get the Qt document
+            doc = view.document
+
+            # get closest node for ins
+            new_text_pos = doc.find_closest_node_pos(curr_ins)
+
+            if new_text_pos is not None:
+                # set the new cursor position
+                textedit = view.textedit
+                cursor = textedit.textCursor()
+                cursor.setPosition(new_text_pos)
+                textedit.setTextCursor(cursor)
+                textedit.setFocus()
+
+
+    def decompile_function(self, func, curr_ins, view=None):
```

Comment:
If you are changing the prototype of a method, you should change the docstring and type-annotate the arguments.

Reply:
added typing and better docstring.